### DISCUSSION TITLE
BACnet Entities should be available as Energy devices in the HA Energy Dashboard

### DISCUSSION THREAD

**Discussion by CV-Private** [Bepacom-Raalte/bepacom-HA-Addons#62] [Ideas]
BACnet Entities should be available as Energy devices in the HA Energy Dashboard.
I can currently not select any entity of my BACnet device as an energy meter in the Energy dashboard, which is a pity, because information about power consumption is available from BACnet objects, but, apparently, the corresponding entities are not meeting the requirements for being recognized as Energy sensors.
In the HA Help, I found this information page about the reason: https://www.home-assistant.io/docs/energy/faq/#troubleshooting-missing-entities 
However, as the BACnet entities are generated by the Bepacom Integration, I don't know any way how I could possibly configure entities so that they would show up as Energy devices.

**Comment by GravySeal** [comment]
Easiest way to get an entity into the right device class, is to create a helper called "Template Sensor" like so.

![image](https://private-user-images.githubusercontent.com/93523925/324174823-fc467331-6f1b-466d-976f-ef89fd91c185.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUxMDk3NTUsIm5iZiI6MTcxNTEwOTQ1NSwicGF0aCI6Ii85MzUyMzkyNS8zMjQxNzQ4MjMtZmM0NjczMzEtNmYxYi00NjZkLTk3NmYtZWY4OWZkOTFjMTg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA3VDE5MTczNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE3NDY3OGJhNWI2ZmYzMmE2YjIyY2MzMjI1YTQyMTQzMTliMWEwNDdmMGRjNGQwZDcwYThhNTZmNDA2OTVmNzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UhX1mttarECSs-53uBYpNFl3a68YV-aCvDHbwYkG3t8)

Here you can choose device class "Energy" as well as state class total for it to show up in the energy menu.
The latest integration does attempt to assign a device class based on units, but as units can be used in multiple device classes this doesn't always result in the right assignment. 
Assigning based on object description has been considered, but as every language results in a different description, it's not a great idea.

Hope this helps!

**Reply by CV-Private** [reply]
Thanks, I'll try that. Could you please paste the image again? I just see a blank page under the link above.

**Reply by GravySeal** [reply]
Yes of course, here's a new screenshot.

![image](https://github.com/Bepacom-Raalte/bepacom-HA-Addons/assets/93523925/28c9afd2-7e16-4ee4-a90f-12665d1261c0)
